enzyme/Enzyme/AdjointGenerator.h

```diff
@@ -8068,14 +8127,34 @@ class AdjointGenerator
             IRBuilder<> Builder2(&call);
             getForwardBuilder(Builder2);
 
-            Value *vdiff = diffe(orig->getArgOperand(0), Builder2);
+            Value *diff = diffe(orig->getArgOperand(0), Builder2);
             Value *exponent =
                 gutils->getNewFromOriginal(orig->getArgOperand(1));
 
-            Value *args[] = {vdiff, exponent};
+            auto rule = [&](Value *diff) {
+              if (auto vty = dyn_cast<VectorType>(diff->getType())) {
+#if LLVM_VERSION_MAJOR >= 12
+                unsigned ec = vty->getElementCount().getKnownMinValue();
+#else
+                unsigned ec = vty->getNumElements();
+#endif
+                Value *vdiff = UndefValue::get(vty);
+                for (int i = 0; i < ec; i++) {
```

Comment:
Is there a reason why this isn't already handled by the applyChainRule?

Reply:
no